packages/nextjs/hooks/scaffold-eth/useTargetNetwork.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from "react";
+import { useNetwork } from "wagmi";
+import scaffoldConfig from "~~/scaffold.config";
+import { useGlobalState } from "~~/services/store/store";
+import { ChainWithAttributes } from "~~/utils/scaffold-eth";
+import { NETWORKS_EXTRA_DATA } from "~~/utils/scaffold-eth";
+
+export function useTargetNetwork(): { targetNetwork: ChainWithAttributes } {
+  const { chain } = useNetwork();
+  const { targetNetwork, setTargetNetwork } = useGlobalState(({ targetNetwork, setTargetNetwork }) => ({
+    targetNetwork,
+    setTargetNetwork,
+  }));
+
+  useEffect(() => {
+    const newSelectedNetwork = scaffoldConfig.targetNetworks.find(targetNetwork => targetNetwork.id === chain?.id);
+    if (newSelectedNetwork && newSelectedNetwork.id !== targetNetwork.id) {
+      setTargetNetwork(newSelectedNetwork);
+    }
+  }, [chain, setTargetNetwork, targetNetwork.id]);
```

Comment:
maybe we do `chain?.id` ?

Reply:
Don't think it matter here (I don't believe the reference to `chain` will ever change without the `id` changing).

But I'll update it for consistency :blush: